### ISSUE TITLE
[MNG-3658] improve maven embedder documentation

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#5365]
**[Benjamin Pochat](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bpochat)** opened **[MNG-3658](https://issues.apache.org/jira/browse/MNG-3658?redirect=false)** and commented

Hi,

I would like to use the maven embedder.
What is described in the maven website seems wonderfull (http://maven.apache.org/guides/mini/guide-embedding-m2.html)
But when I tried to go further, I was not able to find a precise documentation about the API that seems to be provided.

The documentation that comes out from Google (http://repo1.maven.org/maven2/org/apache/maven/maven-embedder/2.0.4/) looks like out of date : maven website describes that the class MavenEmbedder  implements a method "execute" returning a MavenExecutionResult ; but this method is not documented in the javadoc coming out from Google.

It even looks like the maven embedder is not included in the standard distribution of maven : there is no classe named MavenEmbedder in the lib maven-core-2.0.X.jar, neither in the dependecies of the pom maven-core-2.0.X.pom.

In other words, I have nothing to make maven embedding work.

Regards,

Benjamin

---

**Affects:** 2.0.7

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

there is no longer a 2.0.x version
